iceoryx_hoofs/source/cxx/command_line_parser.cpp

```diff
@@ -0,0 +1,520 @@
+// Copyright (c) 2022 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "iceoryx_hoofs/internal/cxx/command_line_parser.hpp"
+#include "iceoryx_hoofs/cxx/algorithm.hpp"
+#include "iceoryx_hoofs/error_handling/error_handling.hpp"
+
+#include <algorithm>
+
+namespace iox
+{
+namespace cxx
+{
+CommandLineParser::CommandLineParser(const description_t& programDescription) noexcept
+    : m_programDescription{programDescription}
+{
+    std::move(*this).addOption({'h', {"help"}, {"Display help."}, ArgumentType::SWITCH, {""}, {""}});
+}
+
+bool CommandLineParser::hasArguments(const int argc) const noexcept
+{
+    const bool hasArguments = (argc > 0);
+    if (!hasArguments)
+    {
+        printHelpAndExit("empty");
+    }
+    return hasArguments;
+}
+
+bool CommandLineParser::assignBinaryName(const char* name) noexcept
```

Comment:
I would not do this.
1. Those functions are all used internally only
2. As long as you are in range with `argc` `argv` will never contain a `nullptr`
3. All those functions are just created so that `parse` becomes more readable. But this has the side effect that you should never use them outside of the `parse` context since the order of the function calls ensure that the preconditions are satisfied.

In the end this is the trade off of making this code more readable. But I am open for suggestions.

Would you be alright to refrain from a `nullptr` check under these circumstances?

Reply:
Okay, I think this is fine.